web/libs/editor/src/components/Comments/Comment/CommentItem.tsx

```diff
@@ -42,198 +45,227 @@ interface CommentItemProps {
-                  No
-                </Button>
-              </Elem>
+        );
+      return null;
+    };
+
+    return (
+      <Block
+        name="comment-item"
+        mod={{ resolved, highlighted: isHighlighted }}
+        onMouseEnter={() => {
+          setHighlighted(true);
+        }}
+        onMouseLeave={() => {
+          setHighlighted(false);
+        }}
+        ref={_commentRef}
+      >
+        <Space spread size="medium" truncated>
+          <Space size="small" truncated>
+            <Elem tag={Userpic} user={createdBy} name="userpic" showUsername username={createdBy} />
+            <Elem name="name" tag="span">
+              {userDisplayName(createdBy)}
             </Elem>
-          ) : (
-            <>
-              {classifications?.default?.values?.length > 0 && (
-                <Elem name="classifications" tag="ul">
-                  {classifications?.default?.values?.map((valueArray: string[], index: number) => (
-                    <li key={index}>{valueArray.join("/")}</li>
-                  ))}
-                </Elem>
-              )}
-              {text}
-              {hasLinkState && (
-                <Elem name="linkState">
-                  <LinkState linking={linking} region={region} result={result} interactive />
+          </Space>
+
+          <Space size="small">
+            <Elem name="resolved" component={IconCheck} />
+            <Elem name="saving" mod={{ hide: isPersisted }}>
+              <Elem name="dot" />
+            </Elem>
+            <TimeTracker />
+          </Space>
+        </Space>
+
+        <Elem name="content">
+          <Elem name="text">
+            {isEditMode ? (
+              <>
+                <CommentFormBase
+                  value={text}
+                  onSubmit={async (value) => {
+                    await updateComment(value, classifications);
+                    setText(value);
+                    await listComments({ suppressClearComments: true });
+                  }}
+                  classifications={classifications}
+                />
+                {classificationsItems.length > 0 && (
+                  <Elem name="classifications-row">
+                    <Taxonomy
+                      selected={taxonomyPathsToSelectedItems(classifications?.default?.values)}
+                      items={classificationsItems}
+                      onChange={async (_: Node, values: TaxonomyPath[]) => {
+                        const newClassifications =
+                          values.length > 0
+                            ? {
+                                default: {
+                                  type: "taxonomy",
+                                  values,
+                                },
+                              }
+                            : null;
+                        setClassifications(newClassifications);
+                      }}
```

Comment:
This should be extracted to a function with a useCallback wrapping it, otherwise there is a chance that on each keystroke the Taxonomy would be re-rendered since this function will change each render of the parent, influenced by all other siblings as well.

Reply:
Is it enough to add just `setClassifications` to the callback's depenendencies?

Reply:
Yes, that is the only function which is used within. I went into more depth above about what should be included, but a good default is to just include the things used, unless it is ref or externally defined function that has no component scope dependencies.